client/src/components/ContentNavigator/convert.ts

```diff
@@ -0,0 +1,241 @@
+      UI_NOTE_PROP_WIDTH: "0",
+    },
+  },
+  priority: 0,
+  properties: {
+    UI_PROP_COLORGRP: "0",
+    UI_PROP_IS_INPUT_EXPANDED: "false",
+    UI_PROP_IS_OUTPUT_EXPANDED: "false",
+    UI_PROP_NODE_DATA_ID: null,
+    UI_PROP_NODE_DATA_MODIFIED_DATE: null,
+    UI_PROP_XPOS: "0",
+    UI_PROP_YPOS: "75",
+  },
+  portMappings: [
+    {
+      mappingType: "tableStructure",
+      portIndex: 0,
+      portName: "outTables",
+      tableStructure: {
+        columnDefinitions: null,
+      },
+    },
+  ],
+  stepReference: {
+    type: "uri",
+    path: null,
+  },
+  arguments: {
+    codeOptions: {
+      code: null,
+      contentType: "embedded",
+      logHTML: "",
+      resultsHTML: "",
+      variables: [
+        {
+          name: "_input1",
+          value: {
+            portIndex: 0,
+            portName: "inTables",
+            referenceType: "inputPort",
+          },
+        },
+        {
+          name: "_output1",
+          value: {
+            arguments: {},
+            portIndex: 0,
+            portName: "outTables",
+            referenceType: "outputPort",
+          },
+        },
+      ],
+    },
+  },
+};
+
+function generateFlowData(inputList: Entry[], outputFile: string) {
+  const now = new Date();
+  const nowString = now.toISOString();
+  const nowTimestamp = String(now.getTime());
+  const arrayIdNode: string[] = [];
+
+  const flowData = { ...baseFlow };
+  flowData.creationTimeStamp = nowString;
+  flowData.modifiedTimeStamp = nowString;
+  flowData.name = outputFile;
+
+  for (let idx = 0; idx < inputList.length; idx++) {
+    const idNode = v4();
+    const idNote = v4();
+    const entry = inputList[idx];
+    const propPorts = getPropPort(idx, inputList);
+
+    const stepNode = {
+      ...baseNode,
+      id: idNode,
+      name: stepTitle[entry.language],
+      note: {
+        ...baseNode.note,
+        id: idNote,
+      },
+      priority: idx,
+      properties: {
+        ...baseNode.properties,
+        ...propPorts,
+        UI_PROP_NODE_DATA_ID: stepRef[entry.language],
+        UI_PROP_NODE_DATA_MODIFIED_DATE: nowTimestamp,
+        UI_PROP_XPOS: ((idx + 1) * NODE_SPACING).toString(),
+      },
+      stepReference: {
+        ...baseNode.stepReference,
+        path: `/dataFlows/steps/${stepRef[entry.language]}`,
+      },
+      arguments: {
+        ...baseNode.arguments,
+        codeOptions: {
+          ...baseNode.arguments.codeOptions,
+          code: entry.code,
+        },
+      },
+    };
+
+    flowData.nodes[idNode] = stepNode;
+    arrayIdNode.push(idNode);
+
+    if (idx > 0) {
+      const connection = {
+        sourcePort: {
+          node: arrayIdNode[idx - 1],
+          portName: "outTables",
+          index: 0,
+        },
+        targetPort: {
+          node: arrayIdNode[idx],
+          portName: "inTables",
+          index: 0,
+        },
+      };
+      flowData.connections.push(connection);
+    }
+  }
```

Comment:
I'm guessing this will change when we move to using swimlanes and _not_ creating connections?

Reply:
Added a new default conversion mode in user settings. Default is **swimlane** conversion but user can switch to **node** conversion. Added in 8cb6a7941c05fbc3634489bca0abfb5609c49876